app/Http/routes.php

```diff
@@ -4,15 +4,31 @@
 |--------------------------------------------------------------------------
 | API Routes - Pages
 |--------------------------------------------------------------------------
-
 */
+Route::get('/developer', [
+    'uses' => 'ApiPagesController@index',
+    'as' => 'developer.index',
+]);
 
-Route::get('/developer', 'ApiPagesController@index');
-Route::get('/developer/myapp', 'ApiPagesController@myApp');
-Route::get('/developer/myapp/new', 'ApiPagesController@createNewApp');
-Route::post('/developer/myapp/new/', function () {
-    return "App created successfully";
-});
+Route::get('/developer/myapp', [
+    'uses' => 'ApiPagesController@showMyApps',
+    'as' => 'developer.myapp',
+]);
+
+Route::get('/developer/myapp/new', [
+    'uses' => 'ApiPagesController@createNewApp',
+    'as' => 'developer.new-app',
+]);
+
+Route::post('/developer/myapp/new/', [
+    'uses' => 'ApiPagesController@postNewAppDetails',
+    'as' => 'developer.save-app',
```

Comment:
**Code Quality - Critical**

It's not correct to have two different named routes for the same route. As long as one has been defined for one of the http verbs, that one applies to the  other routes. e.g.

```php

Route::get('/developer/myapp/new', [
 'uses' => 'ApiPagesController@createNewApp',
 'as'    => 'developer.new-app'
]);

Route::post('/developer/myapp/new', [
 'uses' => 'ApiPagesController@postNewAppDetails'
]);

Route::put('/developer/myapp/new', [
 'uses' => 'ApiPagesController@updateNewAppDetails'
]);
````

This is how it works. So please eliminate the other named routes for this particular route and follow the example I just gave.

Also, check other routes and resolve the mistakes.